crates/sui-indexer-alt-framework/src/pipeline/processor.rs

```diff
@@ -138,3 +138,204 @@ pub(super) fn processor<P: Processor + Send + Sync + 'static>(
         };
     })
 }
+
+#[cfg(test)]
```

Comment:
Could we add a test that tests out the concurrency of the processor? This concurrency can be controlled by the `FANOUT` constant in the processor impl, which defaults to 10. I'm curious to see a test case where the number of checkpoints in the stream exceeds the fanout of the processor.